test/index.js

```diff
@@ -372,6 +372,18 @@ describe('mongoose-patch-history', () => {
         .catch(done)
     })
 
+    it('should not throw "TypeError: Cannot set property _original of null" error with options { new:true } if doc does not exist', (done) => {
+      Post.findOneAndUpdate(
+        { title: 'the_answer_to_life' },
+        { title: '42', comments: 'thanks for all the fish' },
+        { new: true }
+      )
+        .then(() => {
+          done()
+        })
+        .catch(done)
+    })
+
     it('should work with options { new:true, rawResult:true }', (done) => {
```

Comment:
I would add another test here to check for the correct behaviour of `should work with options { new:true }` since the code responsible for this option got changed.

```js
    it('should work with options { new:true }', (done) => {
      Post.findOneAndUpdate(
        { title: 'findOneAndUpdate1' },
        { title: 'findOneAndUpdate2' },
        { new: true }
      )
        .then(() => done())
        .catch(done)
    })
```

Reply:
I added this test case.

Reply:
Should we make this test case more thorough? I am thinking about querying the generated patches if any.

Sorry for requesting this as a second change and not in the previous review.
We could also do this at a later stage. Depends on your workload.

Thanks in advance and thanks for the changes!

Reply:
IMO there is no result in this test case. But I agree with you that the tests should be more specific.